configure.ac

```diff
@@ -166,7 +166,7 @@ PHP_RUNPATH_SWITCH
 dnl Checks for some support/generator progs.
 PHP_PROG_AWK
 PHP_PROG_BISON([3.0.0])
-PHP_PROG_RE2C([1.0.3])
+PHP_PROG_RE2C([1.2.1])
```

Comment:
I think there is a reason this was never bumped. I remember from some time ago that there were issues with the date/time parser with newer versions of `re2c`. The version for that must stay on 1.0.3. That shouldn't be a problem though, as it isn't generated on-demand, but rather the generated files are committed.

I am still weary of this change.

Reply:
@derickr thanks for the feedback. I'll give it a few more tries with the tools that are available on 1.0.3, but this is my first non-trivial change to re2c and it's possible I'm missing something.

If I (or anybody else) cannot come up with a better solution, one option could be to put support for MySQL's space-after  `--` comment requirement (which is the only reason for this) as a secondary vote, perhaps at 2/3 majority too, given the risky requirement.

Reply:
@derickr After some fiddling, bf3aa65 manages to achieve the same result without requiring a new version of re2c.

TL;DR
The whitespace requirement after `--` turns some `YYFILL(2)` calls into `YYFILL(3)` in the generated C file. The unwanted effect result was returning with `PDO_PARSER_EOI` earlier than expected and skipping e.g. a `PDO_PARSER_BIND_POS` when it was the last character in the input string.